### DISCUSSION TITLE
Bug in AI Autoembeddings when creating multiple autoembeddings

### DISCUSSION THREAD

**Discussion by osseonews** [nhost/nhost#2645] [Q&A] [answered]
Unless I am missing something, you have a bug in AutoEmbeddings. A graphite search will only be created for one Autoembedding. If you create a 2nd autoembedding for a 2nd table (AI->Autoembeddings), it will run but no Graphite Search is created to actually query this autoembedding. In our particular case, we created 2 Autoembeddings for 2 different tables. We know they are working as the autoembeddings get created on schedule without any problems in both tables. However, we currently only see 1 graphiteSearch for the first table (and the corresponding graphite_search functions). For the 2nd table, the embeddings are created, but no graphiteSearch is created.

**Comment by osseonews** [comment]
In our logs we see the following error (ids changed) when trying to create an autoembedding. We have tried this many times. Nothing works. 

```js

{"detail":{"http_info":{"content_encoding":null,"http_version":"HTTP/1.1","ip":"","method":"POST","status":400,"url":"/v2/query"},"operation":{"error":{"code":"postgres-error","error":"query execution failed","internal":{"arguments":[],"error":{"description":null,"exec_status":"FatalError","hint":"No operator matches the given name and argument types. You might need to add explicit type casts.","message":"operator does not exist: text <=> text","status_code":"42883"}
--
```

```js
method":"POST","url":"/v1/webhooks/auto-embeddings-configuration","errors":["error synchronizing auto embeddings configuration: failed to deploy similar functionality: failed to deploy similar functionality: failed to run SQL: request failed with status 400: {\"error\":\"query execution failed\",\"path\":\"$\",\"code\":\"postgres-error\",\"internal\":{\"arguments\":[],\"error\":{\"description\":null,\"exec_status\":\"FatalError\",\"hint\":\"No operator matches the given name and argument types. You might need to add explicit type casts
```

`{"time":"2024-04-07T19:11:45.544323571Z","level":"ERROR","msg":"failed to run SQL","trace":{"trace_id":"8****","span_id":"1ca73052441d9fad","parent_span_id":"45d4cac152eee166"},"workflow_data":{"name":"questions","id":"ADSFSDAFA","remove":false,"function":"graphite_similar_questions","error":"request failed with status 400: {\"error\":\"query execution failed\",\"path\":\"$\",\"code\":\"postgres-error\",\"internal\":{\"arguments\":[],\"error\":{\"description\":null,\"exec_status\":\"FatalError\",\"hint\":\"No operator matches the given name and argument types. You might need to add explicit type casts.\",\"message\":\"operator does not exist: text <=> text\",\"status_code\":\"42883\"}`

**Comment by dbarrosop** [comment]
would you mind sharing your project's subdomain?

**Reply by osseonews** [reply]
You have a support email or something to send this to? Prefer not to share this here. Anyway, there is certainly some sort of bug here. Our autoembeddings get computed, but the graphite functions never get created. Only the first one we did got created. Subsequent embeddings fail with the error above (even if the "cron" works and the embeddings get created)

**Reply by osseonews** [reply]
If it helps you can search your logs by our company name: dalicommerce. The initial error happens on the creation of an autoembedding, and subsequent errors happen because no graphite functions actually got created, even if there is no bug in any of the information entered (the embeddings part still works, just no graphite functions).

**Reply by dbarrosop** [reply]
You can use either discord or support@nhost.io. As you prefer.

**Reply by osseonews** [reply]
ok just sent to support.

**Comment by osseonews** [comment, marked as answer]
This issue was solved with support. Suggest you improve the documentation for creating autoembeddings. Specifically, when creating the autoembedding, you have a field called: "Column". It is not clear what column in the database this is referring to.  I recommend when you hover over that field, the question mark says: "Exact name of the column in your database that stores the embeddings." Something like that. Thanks.
